packages/driver/cypress/integration/e2e/multi-domain/navigation_spec.ts

```diff
@@ -237,3 +237,253 @@ describe('delayed navigation', { experimentalSessionSupport: true, defaultComman
+    cy.on('fail', (err) => {
+      expect(err.message).to.include(`Timed out after waiting \`5000ms\` for your remote page to load on origin(s):`)
+      expect(err.message).to.include(`\n\`http://localhost:3500\`\n`)
+      expect(err.message).to.include(`Your page did not fire its \`load\` event within \`5000ms\`.`)
+      expect(err.message).to.include(`A cross origin request for \`http://www.foobar.com:3500/fixtures/auth/idp.html?redirect=http%3A%2F%2Flocalhost%3A3500%2Ffixtures%2Fauth%2Findex.html\` was detected.`)
+      expect(err.message).to.include(`To handle cross origin requests, they must be immediately followed by a switchToDomain command:`)
+      expect(err.message).to.include(`\`cy.switchToDomain('http://www.foobar.com:3500', () => { <cross origin commands go here> })\``)
+      expect(err.message).to.include(`If the cross origin was an intermediary state, You can try increasing the \`pageLoadTimeout\` value in \`cypress.json\` to wait longer`)
+
+      expect(err.message).not.to.include(`The following error originated from your test code, not from Cypress`)
+      expect(err.message).not.to.include(`The following error originated from your application code, not from Cypress`)
+      done()
+    })
+
+    cy.visit('/fixtures/auth/index.html')
+    cy.get('[data-cy="login-foobar"]').click() // Timeout on page load here, we never reach the expected domain
+    cy.get('[data-cy="login-foobar"]')
+  })
+
+  it('never redirects to the subdomain', { defaultCommandTimeout: 50 }, (done) => {
+    cy.on('fail', (err) => {
+      expect(err.message).to.include(`Timed out retrying after 50ms: Expected to find element: \`[data-cy="username"]\`, but never found it`)
+      //  make sure that the secondary domain failures do NOT show up as spec failures or AUT failures
+      expect(err.message).not.to.include(`The following error originated from your test code, not from Cypress`)
+      expect(err.message).not.to.include(`The following error originated from your application code, not from Cypress`)
+      done()
+    })
+
+    cy.visit('/fixtures/auth/index.html')
+    cy.get('[data-cy="login-idp"]')
+    cy.switchToDomain('http://idp.com:3500', () => {
+      cy.get('[data-cy="username"]').type('BJohnson') // Timeout here on command, cannot find element
+      cy.get('[data-cy="login"]').click()
+    })
+
+    cy.get('[data-cy="welcome"]')
+    .invoke('text')
+    .should('equal', 'Welcome BJohnson')
+  })
+
+  it('redirects to the wrong subdomain', { pageLoadTimeout: 5000 }, (done) => {
+    cy.on('fail', (err) => {
+      expect(err.message).to.include(`Timed out after waiting \`5000ms\` for your remote page to load on origin(s):`)
+      expect(err.message).to.include(`\n\`idp.com\`\n`)
+      expect(err.message).to.include(`A cross origin request for \`http://www.foobar.com:3500/fixtures/auth/idp.html?redirect=http%3A%2F%2Flocalhost%3A3500%2Ffixtures%2Fauth%2Findex.html\` was detected.`)
```

Comment:
Can we use `decodeURIComponent` or something similar to make the url a bit easier to read in the error message?

Reply:
I don' t know if we should, I'm trying to present the url in the request just as it would be seen in the nav bar.

Reply:
That's what I figured. Sounds like it would be a tradeoff of accuracy for readability. My guess in this scenario, accuracy is likely more preferable!